src/Token.php

```diff
@@ -26,6 +26,11 @@
     public function headers(): DataSet;
 
     /**
+     * Returns the token claims
+     */
+    public function claims(): DataSet;
```

Comment:
> You mean if JWE using, then it must be decrypted first? Or why claims must not be available? And how to get user information from token, if we have not claims?

@Furdarius exactly, if you need to extract information from a nested token (and the information is "public") you can use the headers, otherwise you just decrypt it.

I didn't add that method to the interface because we have plans to support [nested tokens](https://tools.ietf.org/html/rfc7519#appendix-A.2) in the near future, which means that claims are not available for that kind of token.

> Btw, if looks at claims method in Token\Plain class, it refer to inheritdoc which should be in TokenInterface: https://github.com/lcobucci/jwt/blob/master/src/Token/Plain.php#L63
But now you can't find this method in interface.

Thanks for pointing that out 😉